42manito/src/components/Banners/TopBanner.tsx

```diff
@@ -0,0 +1,81 @@
+import React, { useEffect, useState } from "react";
+import Image from "next/image";
+import { useRouter } from "next/router";
+import { BannerDto } from "@/Types/Banners/Banner.dto";
+import { clear } from "console";
+
+interface props {
+  banner: BannerDto[];
+}
+
+// 추후 여러 배너를 받을 수 있는 형태로 수정이 필요함.
+export default function TopBanner({ banner }: props) {
+  const router = useRouter();
+  const [currentIndex, setCurrentIndex] = useState(0);
+  const handleBannerClick = () => {
+    router.push(banner[currentIndex].link);
+  };
+  var timerId: any;
+  const timer = 5000; // 임의로 넣어뒀습니다
+
+  const autoPlay = () => {
+    timerId = setTimeout(() => {
```

Comment:
ESLint: Assignments to the 'timerId' variable from inside React Hook useEffect will be lost after each render. To preserve the value over time, store it in a useRef Hook and keep the mutable value in the '.current' property. Otherwise, you can move this variable directly inside useEffect.(react-hooks/exhaustive-deps)

eslint 에러가 나옵니다 time id 를 let이나 var를 통한 변수가 아니라 상태로 접근해야하는 것 같아요.

Reply:
log 지우고 state로 관리하도록 수정했습니다 확인 부탁드려여~~